src/ListView/CreatePage/CreatePageForm.jsx

```diff
@@ -0,0 +1,132 @@
+import React, { useEffect, useState } from "react";
+import { colorTextErrorOnLight } from "@sailthru/design-tokens";
+import {
+  FormGroup,
+  Radio,
+  RadioGroup,
+  Subtext,
+} from "@sailthru/stui-components";
+import { Fieldset, Label, Input } from "@sailthru/stui-elements";
+
+function CreatePageForm({
+  name = "",
+  onChange,
+  isDuplicateName,
+  isNameChanged,
+  setIsValidName,
+}) {
+  const [namingError, setNamingError] = useState("");
+  const [optoutNamingError, setOptoutNamingError] = useState("");
+
+  const checkForNamingError = function () {
+    setNamingError("");
+    setOptoutNamingError("");
+    if (name.length < 1) {
+      setNamingError("Please enter a page name.");
+      setIsValidName(false);
+      return;
+    }
+    if (name === "oc") {
+      setNamingError(
+        "The name 'oc' is a reserved page name. Please use a different page name."
+      );
+      setIsValidName(false);
+      return;
+    }
+    if (name === "optout") {
+      setOptoutNamingError(
+        "The page name 'optout' requires 'User Management' as the category. Please select 'User Management' below."
+      );
+      return;
+    }
+    const alphanumericRegex = /^[a-zA-Z0-9]+$/;
+    const nameWithoutDashes = name.replace(new RegExp("-", "g"), "");
+    if (!nameWithoutDashes.match(alphanumericRegex)) {
+      setNamingError(
+        "The page name must contain at least one alphanumeric character and may contain dashes. Please use a different page name."
+      );
+      setIsValidName(false);
+      return;
+    }
+  };
+
+  useEffect(() => {
+    if (isDuplicateName) {
+      setNamingError(
+        "The page name must be unique. Please use a different page name."
+      );
+    }
+  }, [isDuplicateName]);
+
+  const onChangeHandler = (e) => {
+    if (isNameChanged) {
+      setNamingError("");
+      setOptoutNamingError("");
+      setIsValidName(true);
+    }
+    onChange({ name: e.target.value });
+  };
+
+  const onOptoutChangeHandler = (e) => {
+    if (name === "optout") {
+      setOptoutNamingError(" "); //this must be a space and not an empty string or the error will clear
+    }
```

Comment:
I'm confused by this use case. Can you explain more why the space is needed? Why not put an actual error message? Does this negation do the same thing?
```suggestion
    if (name !== "optout") {
      setOptoutNamingError("");
    }
```

Reply:
yes it does!